core/saltfilter/bloom.go

```diff
@@ -0,0 +1,34 @@
+// Package saltfilter used for reuse detection ONLY
+package saltfilter
+
+import (
+	"sync"
+
+	"github.com/willf/bloom"
+)
+
+var (
+	filterInstance *bloom.BloomFilter
+	itemCount      = 0
+	mux            = sync.Mutex{}
+)
+
+const itemSize = 1e6
```

Comment:
Rather than hardcoding this value, it's better to make it configurable via `flag` so people can adjust the size according to their memory budget for security.